docs/coding-guidelines/typescript-dev-flow.dot

```diff
@@ -0,0 +1,34 @@
+digraph {
+	label="What are my expecations when working with TypeScript?";
+
+    node[shape="box", style="rounded"]
+        existing_file; new_file;
+
+    node[shape="diamond"]
+        am_comfortable; am_comfortable_;
+
+    node[shape="parallelogram"]
+        no_types; learning; learning_; love_types;
+
+    existing_file -> am_comfortable;
+
+    am_comfortable -> no_types[label="no"];
+    am_comfortable -> learning[label="no but I want to be"];
+    am_comfortable -> love_types[label="yes"];
+
+    existing_file[label="opening an existing file"];
+    am_comfortable[label="am I comfortable\nworking with strong\nstatic types?"];
+    no_types[label="don't sweat it.\nwrite normal JavaScript in the\nfile as much as you can."];
+    learning[label="add as much or little TypeScript as you want\n\nthe types tribe will review your work\nand provide feedback to help you learn"];
```

Comment:
Can you rename the "types tribe" references to maybe "type review team", something that better aligns with the team we would tag for help?

Reply:
Also picky: can you capitalize the first letter of sentences?

Reply:
updated and rebased so the actual commits are somewhat hidden